electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -431,11 +432,38 @@ def append(
             capacity_reduction,
             fuel_type,
             outage_type,
+            generator_id,
             reason,
         )
         if event:
             self.events.append(event)
 
+    @staticmethod
+    def aggregate_across_generation_units(
+        outages: "OutageList", logger: Logger
+    ) -> "OutageList":
+        """
+        Aggregates the outages over the generation units.
+        """
+        aggregated_outages_df = (
+            pd.DataFrame([o.to_dict() for o in outages])
+            .drop_duplicates()
+            .groupby(["datetime", "zoneKey", "source", "fuel_type", "outage_type"])
+            .sum(numeric_only=True)
+            .sort_index()
+            .reset_index()
+        )
+        aggregated_outages = OutageList(logger)
+        for _, row in aggregated_outages_df.iterrows():
+            aggregated_outages.append(
+                zoneKey=row["zoneKey"],
+                datetime=row["datetime"],
+                source=row["source"],
+                capacity_reduction=row["capacity_reduction"],
+                fuel_type=row["fuel_type"],
+            )
```

Comment:
We have in the past tried to avoid converting to dataframes in the parser models as usually the performance overhead mad maintenance cost is higher than the benefit. (Especially we try to update dependencies).

Is there a way to do this without pandas in a easy way?